### PR TITLE
ISPN-15202 Mask Hot Rod and database credentials when serializing con…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-15202

### CODE REVIEW

File: commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSerializer.java [hunk 1]
```diff
@@ -20,15 +20,22 @@ public interface AttributeSerializer<T> {
...
    };
-   AttributeSerializer<Supplier<char[]>> SECRET = (writer, name, value) -> {
+   AttributeSerializer<String> SECRET = (writer, name, value) -> {
+      if (writer.clearTextSecrets()) {
+         writer.writeAttribute(name, value);
+      } else {
+         writer.writeAttribute(name, "***");
+      }
+   };
+   AttributeSerializer<Supplier<char[]>> SECRET_SUPPLIER = (writer, name, value) -> {
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSerializer.java:30]: Is this ever used?

Comment (tristantarrant) [reply commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSerializer.java]: I've removed it

**Comment by karesti** [comment]
@tristantarrant I tried making work an example with the. branch locally, but is a bit long. So I'm just wondering, is the REST API config also being masked by these changes? on the config side of things for admin users

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Comment by ryanemerson** [comment]
Don't forget to backport to 14.0.x :slightly_smiling_face:

**Pull request merged**